### PR TITLE
Rework the README and instruct users to run the generator

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>
/assign @petr-muller

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
+and are split into the following categories:
+
+ - [`ci-operator/config`](./ci-operator/config/) contains configuration for the
+   `ci-operator`, detailing builds and tests for component repositories. See the
+   [contributing guide](./ci-operator/README.md) for details on how to configure
+   new repositories or tests.
+ - [`ci-operator/jobs`](./ci-operator/jobs/) contains configuration for `prow`,
+   detailing job triggers. See the [upstream configuration document](https://github.com/kubernetes/test-infra/blob/master/prow/README.md#how-to-add-new-jobs)
+   for details on how to configure a job, but prefer the `ci-operator` config
+   whenever possible.
```
Comment (petr-muller) [inline README.md]: This fails to explain how `ci-operator` config is actually a (better) alternative to manually adding a job. Something like:

... contains configuration for Prow detailing job triggers. Most of this configuration is generated automatically from the ci-operator config. See the ... on how to manually configure a job, but prefer the `ci-operator` config whenever possible.

File: ci-operator/README.md [hunk 2]
```diff
@@ -5,6 +5,20 @@ This document describes how to create CI jobs for Openshift components using
...
+After editing the files under this directory, make sure to run the generator to
+ensure that your changes are compliant to our conventions and pass the CI tests
+that will run when you submit your changes as a PR:
+
+```
+docker pull registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest
+docker run -it -v $(pwd)/ci-operator:/ci-operator:z           \
+  registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest \
+  --from-dir /ci-operator/config/ --to-dir /ci-operator/jobs
+```
```
Comment (petr-muller) [inline ci-operator/README.md]: Instructing people to run the generator is one thing, but we maybe we should advise them what is the expected result to pass the CI later.

Comment (stevekuznetsov) [reply ci-operator/README.md]: What do you mean?

Comment (petr-muller) [reply ci-operator/README.md]: I am an engineer trying to follow these instructions. I make some changes and run dockerized generator. Right now, it spits out ~40 warnings about name length you don't need to care about (side note: I now think we should remove these). The CI will pass if 1) generator does not hit any error, 2) the generator does not generate anything that differs from the committed state. We can expect people to find out 1) is needed, but not 2). So we should tell them.

"Running the generator over your commits should not further change any of the files under `ci-operator/jobs`; if it does, the CI will fail." or something similar.

**Comment by stevekuznetsov** [comment]
@petr-muller updated the PR to be more explicit there, added a PR to the generator to be more intelligent

**Comment by petr-muller** [comment]
Huh, why are all the
```
- - master
+ - ^master$
```

changes here? Was that intentional?

File: ci-operator/README.md [hunk 3]
```diff
@@ -5,6 +5,22 @@ This document describes how to create CI jobs for Openshift components using
...
+
+Make sure to pull the latest image to ensure that you do not use a stale version
+of the generator. Pre-submit tests on this repository will ensure that a run of
+the latest generator does not error on proposed congfiguration changes and also
```
Comment (petr-muller) [inline ci-operator/README.md]: typo: `congfiguration` -> `configuration`

**Comment by stevekuznetsov** [comment]
I ran the generator ... if I remove those, I assume this will fail? It wasn't intentional but is this branch of `openshift/release` broken?

**Pull request merged**